packages/autocomplete-core/src/utils.ts

```diff
@@ -92,35 +92,38 @@ export function getNextHighlightedIndex<TItem>(
 // We don't have access to the autocomplete source when we call `onKeyDown`
 // or `onClick` because those are native browser events.
 // However, we can get the source from the suggestion index.
-function getSuggestionFromHighlightedIndex<TItem>({
+function getCollectionFromSelectedItemId<TItem>({
   state,
 }: {
   state: AutocompleteState<TItem>;
-}): AutocompleteSuggestion<TItem> | undefined {
+}): AutocompleteCollection<TItem> | undefined {
   // Given 3 sources with respectively 1, 2 and 3 suggestions: [1, 2, 3]
   // We want to get the accumulated counts:
   // [1, 1 + 2, 1 + 2 + 3] = [1, 3, 3 + 3] = [1, 3, 6]
-  const accumulatedSuggestionsCount = state.suggestions
-    .map((suggestion) => suggestion.items.length)
-    .reduce<number[]>((acc, suggestionCount, index) => {
+  const accumulatedCollectionssCount = state.collections
```

Comment:
`accumulatedCollectionssCount` -> two `s`s, intended or a typo?

Reply:
Good catch—perhaps that announces a new broken key on my keyboard 🤪

See e162b5158c67a2915c2b972ad28a2f356df735d6.